### PR TITLE
Railroad activity

### PR DESCRIPTION
**Pull request by IrshaadAli**
I have created UI design using flow and also created basic animations, onClick events.

### CODE REVIEW

File: src/activities/railroad/ActivityInfo.qml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* GCompris - ActivityInfo.qml
+ *
+ * Copyright (C) 2015 Your Name <yy@zz.org>
```
Comment (petitlapin) [inline src/activities/railroad/ActivityInfo.qml]: Hi,

please update with your name

File: src/activities/railroad/ActivityInfo.qml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+import GCompris 1.0
+
+ActivityInfo {
+  name: "railroad/Railroad.qml"
+  difficulty: 1
+  icon: "railroad/railroad.svg"
+  author: "Irshaad Ali &lt;irshaadali14@gmail.com   &gt;"
+  demo: true
+  title: "Railroad activity"
```
Comment (petitlapin) [inline src/activities/railroad/ActivityInfo.qml]: use qsTr() for translatable strings

File: src/activities/railroad/ActivityInfo.qml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  description: "A memory game based on trains"
+  //intro: "put here in comment the text for the intro voice"
+  goal: ""
+  prerequisite: ""
+  manual: "A train - a locomotive and carriage(s) - is displayed at the top of the main area for a few seconds. Rebuild it at the top of the screen by selecting the appropriate carriages and locomotive. Deselect an item by clicking on it again. Check your construction by clicking on the hand at the bottom."
+  credit: ""
+  section: "discovery"
+  createdInVersion: 4000
```
Comment (petitlapin) [inline src/activities/railroad/ActivityInfo.qml]: it should be 6000

File: src/activities/railroad/Railroad.qml [hunk 4]
```diff
@@ -0,0 +1,133 @@
+/* GCompris - railroad.qml
+ *
+ * Copyright (C) 2015 YOUR NAME <xx@yy.org>
+ *
+ * Authors:
+ *   <THE GTK VERSION AUTHOR> (GTK+ version)
```
Comment (petitlapin) [inline src/activities/railroad/Railroad.qml]: update copyrights

File: src/activities/railroad/railroad.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
+/* GCompris - railroad.js
+ *
+ * Copyright (C) 2015 YOUR NAME <xx@yy.org>
+ *
+ * Authors:
+ *   <THE GTK VERSION AUTHOR> (GTK+ version)
```
Comment (petitlapin) [inline src/activities/railroad/railroad.js]: copyrights

File: src/activities/railroad/railroad.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            ["wagon11.svg",100,100],
+            ["wagon12.svg",100,100],
+            ["wagon13.svg",100,100],
+
+        ]
+
+
+var url = "qrc:/gcompris/src/activities/railroad/resource/"
+
+if(trainPart.visible == true){
```
Comment (petitlapin) [inline src/activities/railroad/railroad.js]: be careful of indentation. Also this code is out of any function, is it normal?

File: src/activities/railroad/Railroad.qml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            property alias background: background
+            property alias bar: bar
+            property alias bonus: bonus
+            property alias locAndWagonsRepeater: locAndWagonsRepeater
+        }
+
+        Flickable {
+            id: engines
+            clip: false
+            width: background.width ; height: 800
```
Comment (petitlapin) [inline src/activities/railroad/Railroad.qml]: why are 800 and 1180 hardcoded values?

File: src/activities/railroad/TrainPart.qml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.2
+import "railroad.js" as Activity
+import GCompris 1.0
+
+Image {
+    id: trainPart
+    width: 236 ; height: 112
```
Comment (petitlapin) [inline src/activities/railroad/TrainPart.qml]: Is it normal that they are hardcoded? For images, it's preferable to set the sourceSize.{width/height}.

File: src/activities/railroad/TrainPart.qml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+    NumberAnimation on x {
+        id: animation
+        to: 1300
+        duration: 16000
+        onRunningChanged: {
+            if (!animation.running) {
+                trainPart.visible = false
+                engines.visible = true
+                animation.destroy()
```
Comment (petitlapin) [inline src/activities/railroad/TrainPart.qml]: I'm not sure you should destroy the animation here (or anywhere else)

File: src/activities/railroad/railroad.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import QtQuick 2.0 as Quick
+
+var currentLevel = 0
+var numberOfLevel = 3
+var items
+
+var backgroundImage = "railroad-bg.svg"
```
Comment (petitlapin) [inline src/activities/railroad/railroad.js]: unused?

File: src/activities/railroad/railroad.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+
+var trainParts = [
+            ["loco1.svg",100,100],
```
Comment (petitlapin) [inline src/activities/railroad/railroad.js]: what is the use of the two 100?

**Comment by IrshaadAli** [comment]
I have made changes according to your reviews. For image width and height, I used sourceSize. But using this engine size is decries during animation. 
Now still renaming loco and wagon selection part.

File: src/activities/railroad/Railroad.qml [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+                            id : locAndWagons
+                            source: Activity.url + modelData[0]
+
+                        }
+                        MouseArea{
+                            id : railMouseArea
+                            anchors.fill: parent
+                            enabled: rail.mouseEnabled
+                            onClicked:{
+                                Qt.quit()
```
Comment (petitlapin) [inline src/activities/railroad/Railroad.qml]: why :)?

File: src/activities/railroad/Railroad.qml [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+            }
+        }
+
+        onStart: { Activity.start(items) }
+        onStop: { Activity.stop() }
+
+        TrainPart {
+            id: trainpart
+
+            z: 11
```
Comment (petitlapin) [inline src/activities/railroad/Railroad.qml]: do you really need to specify the z index?

File: src/activities/railroad/TrainPart.qml [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+    y: background.width/40
+    source: Activity.url + "loco8.svg"
+    fillMode: Image.PreserveAspectFit
+    z: 10
+
+
+
+    NumberAnimation on x {
+        id: animation
+        to: 1300
```
Comment (petitlapin) [inline src/activities/railroad/TrainPart.qml]: why 1300 hardcoded?

File: src/activities/railroad/railroad.js [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+    items.bar.level = currentLevel + 1
+    console.log(url + trainParts[1])
+}
+
+function nextLevel() {
+    if(numberOfLevel <= ++currentLevel ) {
+        currentLevel = 0
+    }
+    initLevel();
+    trainParts[0][1] = 100
```
Comment (petitlapin) [inline src/activities/railroad/railroad.js]: I'm not sure it is still useful

File: src/activities/railroad/Railroad.qml (hunk 13 above)
Comment (IrshaadAli) [inline src/activities/railroad/Railroad.qml]: I have used z index for some experiment. But right now i don't need z index.

File: src/activities/railroad/railroad.js [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+var currentLevel = 0
+var numberOfLevel = 3
+var items
+
+var backgroundImage = "railroad-bg.svg"
+
+
+var trainParts = [
+            ["loco1.svg"],
```
Comment (bdoin) [inline src/activities/railroad/railroad.js]: Are there really svg images or png in svg. If this is just png in svg, it is best to keep the image in the png format on disk.

**Comment by bdoin** [comment]
On the user interface it looks like the Gtk+ version but it was far from perfect. A drag and drop approach with the ability to insert train parts could be appropriate.

**Pull request closed without merging**